File: static/code/sample/daisylink/gdl-mcpixobit-d.js
// In this sample:
// Draws circles on the screen
// Plays a sweep sound when a touch is detected on pin 0, 1, or 2
// Plays a sweep sound when a pressed is detected on button A,B
// Show light value on the top of the screen

import pkg_serialusb from 'dlserialusb';
const {SerialUSB} = pkg_serialusb

import pkg_duelink from 'duelink';
const {DUELinkController} = pkg_duelink

let duelink = new DUELinkController(new SerialUSB());
await duelink.Connect();


async function IsTouch(pin) {
    if (pin < 0 || pin > 2)
        return false;

    const cmd = `istouch(${pin})`;
    const ret = parseInt(await duelink.Engine.ExecuteCommand(cmd));

    return ret === 1;
}

const buzzerPin = 4;
let x = 50;
let d = 10;
const width = parseInt(await duelink.Engine.ExecuteCommand("getw()"));
const height = parseInt(await duelink.Engine.ExecuteCommand("geth()"));

while (true) {
    await duelink.Graphics.Circle(1, x, Math.floor(height / 2), 10);
    await duelink.Graphics.Show();
    x = x + d;

    if (x > width || x < 0) {
        d = d * -1;
        await duelink.Graphics.Clear(0);
        const light = await duelink.Engine.ExecuteCommand("Light()");
        await duelink.Graphics.TextS(`Light: ${light}%`, 1, 0, 0, 1, 1);
    }

    for (let i = 0; i < 3; i++) {
        if (await IsTouch(i)) {
            await duelink.Sound.Sweep(buzzerPin, 2000, 4000, 50, 255, 100);
            break;
        }
    }

    if (await duelink.Engine.ExecuteCommand("BtnA()") != 0 || await duelink.Engine.ExecuteCommand("BtnB()") != 0) {
        await duelink.Sound.Sweep(buzzerPin, 1000, 2000, 50, 255, 100);
    }

    await new Promise(resolve => setTimeout(resolve, 1)); 
}

File: static/code/sample/tethered/gdl-mcpixobit-d.py
# In this sample:
# Draws circles on the screen
# Plays a sweep sound when a touch is detected on pin 0, 1, or 2
# Plays a sweep sound when a pressed is detected on button A,B
# Show light value on the top of the screen

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

def IsTouch(pin):
    if pin < 0 or pin > 2:
        return False

    cmd = f"istouch({pin})"
    ret = int(duelink.Engine.ExecuteCommand(cmd))

    return ret == 1

buzzerPin = 4
x = 50
d = 10
width = int(duelink.Engine.ExecuteCommand("getw()"))
height = int(duelink.Engine.ExecuteCommand("geth()"))

while True:
    duelink.Graphics.Circle(1, x, height // 2, 10)
    duelink.Graphics.Show()
    x = x + d

    if x > width or x < 0:
        d = d * -1
        duelink.Graphics.Clear(0)
        light = duelink.Engine.ExecuteCommand("Light()")
        duelink.Graphics.TextS(f"Light: {light}%", 1, 0, 0, 1, 1)

    for i in range(3):
        if IsTouch(i):
            duelink.Sound.Sweep(buzzerPin, 2000, 4000, 50, 255, 100)
            break

    if duelink.Engine.ExecuteCommand("BtnA()") != 0 or duelink.Engine.ExecuteCommand("BtnB()") != 0:
        duelink.Sound.Sweep(buzzerPin, 1000, 2000, 50, 255, 100)

    time.sleep(0.001)  





